funding/manager.go

```diff
@@ -1587,9 +1587,34 @@ func (f *Manager) handleFundingAccept(peer lnpeer.Peer,
 			}
 		}
 	} else if msg.ChannelType != nil {
-		err := errors.New("received unexpected channel type")
-		f.failFundingFlow(peer, msg.PendingChannelID, err)
-		return
+		// The spec isn't too clear about whether it's okay to set the
+		// channel type in the accept_channel response if we didn't
+		// explicitly set it in the open_channel message. For now, let's
+		// just log the problem instead of failing the funding flow.
+		log.Warnf("Peer sent explicit channel type even though we " +
```

Comment:
In order to retain backwards compatibility before the feature bit for explicit chan type was added, this is actually ok (so we ca remove the log warning). 

The main thing we need to do is ensure the chan types match up at the end as we do below.